mcf-browser/src/back-end/data.ts

```diff
@@ -0,0 +1,250 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const ID_DELIMITER = ',';
+
+const ID_PROPERTIES = [
+  'variableMeasured',
+  'observationAbout',
+  'provenance',
+  'measurementMethod',
+  'observationPeriod',
+  'unit',
+  'scalingFactor',
+] as const;
+type IdProperties = typeof ID_PROPERTIES[number];
+type SeriesIdObject = Pick<Series, IdProperties>;
+
+type DataPoint = {
+  x: string;
+  y: number;
+};
+
+/** Class representation of a single series */
+class Series {
+  /** A unique identifier made using the facet
+  */
+  id: string;
+
+  /** An ordered list of datapoints */
```

Comment:
nit: ordered by date?  mention

Reply:
Oh it's no longer ordered, I'll remove that